anvil-python/anvil/commands/haproxy.py

```diff
@@ -54,10 +64,34 @@ def __init__(
             "Deploying HAProxy",
             refresh,
         )
+        self.variables = {"virtual_ip": None}
 
     def get_application_timeout(self) -> int:
         return HAPROXY_APP_TIMEOUT
 
+    def has_prompts(self) -> bool:
+        return True
+
+    def prompt(self, console: Console | None = None) -> None:
+        bootstrap_bank = QuestionBank(
+            questions={
+                "virtual_ip": PromptQuestion(
```

Comment:
what about adding the `validation_function` here? https://github.com/canonical/snap-openstack/blob/anvil-0.1/sunbeam-python/sunbeam/jobs/questions.py#L105